barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -209,14 +220,20 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
         auto prev_level_width = prev_level_coeffs.size();
         // we need degree + 1 terms to represent the intermediate polynomials
         std::vector<std::vector<FF>> level_coeffs(prev_level_width >> 1, std::vector<FF>(degree + 1, 0));
-        for (size_t node = 0; node < prev_level_width; node += 2) {
-            auto parent = node >> 1;
-            std::copy(prev_level_coeffs[node].begin(), prev_level_coeffs[node].end(), level_coeffs[parent].begin());
-            for (size_t d = 0; d < degree; d++) {
-                level_coeffs[parent][d] += prev_level_coeffs[node + 1][d] * betas[level];
-                level_coeffs[parent][d + 1] += prev_level_coeffs[node + 1][d] * deltas[level];
-            }
-        }
+        run_loop_in_parallel(
+            prev_level_width >> 1,
+            [&](size_t start, size_t end) {
+                for (size_t node = start << 1; node < end << 1; node += 2) {
+                    auto parent = node >> 1;
+                    std::copy(
+                        prev_level_coeffs[node].begin(), prev_level_coeffs[node].end(), level_coeffs[parent].begin());
+                    for (size_t d = 0; d < degree; d++) {
+                        level_coeffs[parent][d] += prev_level_coeffs[node + 1][d] * betas[level];
+                        level_coeffs[parent][d + 1] += prev_level_coeffs[node + 1][d] * deltas[level];
+                    }
+                }
+            },
+            8);
```

Comment:
will you please add a comment explaining what this 8 is

Reply:
Like this?